library/lib.sh

```diff
@@ -155,15 +175,10 @@ rolesInstallDependencies() {
     for req_file in $coll_req_file $coll_test_req_file; do
         if [ ! -f "$req_file" ]; then
             rlLogInfo "Skipping installing dependencies from $req_file, this file doesn't exist"
-        else
-            rlRun "ansible-galaxy collection install -p $collection_path -vv -r $req_file"
-            if rolesIsAnsibleEnvVarSupported ANSIBLE_COLLECTION_PATH; then
-                rlRun "export ANSIBLE_COLLECTIONS_PATH=$collection_path"
-            else
-                rlRun "export ANSIBLE_COLLECTIONS_PATHS=$collection_path"
-            fi
-            rlLogInfo "$req_file Dependencies were successfully installed"
+            continue
         fi
+        rlRun "ansible-galaxy collection install -p $collection_path -vv -r $req_file"
```

Comment:
do you set the environment variables in `ANSIBLE_ENVS` somewhere before this?  Because on some (all?) versions of `ansible-galaxy`, the `-p` argument is ignored.

Reply:
How's that possible? This worked for me everywhere.

Pasting output of `ansible-galaxy install` from the last test results in https://github.com/linux-system-roles/tft-tests/actions/runs/10909209697/job/30276911937.
CS7:
```
:: [ 14:56:33 ] :: [  BEGIN   ] :: Running 'ansible-galaxy collection install -p /tmp/collections-nIr -vv -r /tmp/postfix-jN7/meta/collection-requirements.yml'
ansible-galaxy 2.9.27
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible-galaxy
  python version = 2.7.5 (default, Nov 14 2023, 16:14:06) [GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]
Using /etc/ansible/ansible.cfg as config file
[WARNING]: The specified collections path '/tmp/collections-nIr' is not part of
the configured Ansible collections paths
'/root/.ansible/collections:/usr/share/ansible/collections'. The installed
collection won't be picked up in an Ansible run.
Process install dependency map
|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/Starting collection install process
|Installing 'containers.podman:1.15.4' to '/tmp/collections-nIr/ansible_collections/containers/podman'
/-\Installing 'ansible.posix:1.6.0' to '/tmp/collections-nIr/ansible_collections/ansible/posix'
|/-Installing 'community.general:9.4.0' to '/tmp/collections-nIr/ansible_collections/community/general'
\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/Installing 'fedora.linux_system_roles:1.88.9' to '/tmp/collections-nIr/ansible_collections/fedora/linux_system_roles'
```

CS8 Ansible 2.9:
```
:: [ 14:56:58 ] :: [  BEGIN   ] :: Running 'ansible-galaxy collection install -p /tmp/collections-r0t -vv -r /tmp/postfix-pIn/meta/collection-requirements.yml'
ansible-galaxy 2.9.27
  config file = None
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.9/site-packages/ansible
  executable location = /usr/local/bin/ansible-galaxy
  python version = 3.9.19 (main, May 16 2024, 11:40:09) [GCC 8.5.0 20210514 (Red Hat 8.5.0-22)]
No config file found; using defaults
[WARNING]: The specified collections path '/tmp/collections-r0t' is not part of
the configured Ansible collections paths
'/root/.ansible/collections:/usr/share/ansible/collections'. The installed
collection won't be picked up in an Ansible run.
Process install dependency map
|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\Starting collection install process
|Installing 'ansible.posix:1.6.0' to '/tmp/collections-r0t/ansible_collections/ansible/posix'
/-\|Installing 'fedora.linux_system_roles:1.88.9' to '/tmp/collections-r0t/ansible_collections/fedora/linux_system_roles'
/-\|/-\|/-\|/-\|/-\|/-\|/-\|/Installing 'community.general:9.4.0' to '/tmp/collections-r0t/ansible_collections/community/general'
-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|Installing 'containers.podman:1.15.4' to '/tmp/collections-r0t/ansible_collections/containers/podman'
```

CS8 Ansible 2.16:
```
:: [ 14:56:47 ] :: [  BEGIN   ] :: Running 'ansible-galaxy collection install -p /tmp/collections-oZe -vv -r /tmp/postfix-8eO/meta/collection-requirements.yml'
[WARNING]: The specified collections path '/tmp/collections-oZe' is not part of
the configured Ansible collections paths
'/root/.ansible/collections:/usr/share/ansible/collections'. The installed
collection will not be picked up in an Ansible run, unless within a playbook-
adjacent collections directory.
ansible-galaxy [core 2.16.11]
  config file = None
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.12/site-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible-galaxy
  python version = 3.12.1 (main, Feb 21 2024, 14:18:26) [GCC 8.5.0 20210514 (Red Hat 8.5.0-21)] (/usr/bin/python3.12)
  jinja version = 3.1.4
  libyaml = True
No config file found; using defaults
Starting galaxy collection install process
Process install dependency map
Starting collection install process
Downloading https://galaxy.ansible.com/api/v3/plugin/ansible/content/published/collections/artifacts/ansible-posix-1.6.0.tar.gz to /root/.ansible/tmp/ansible-local-7695t4x0lor3/tmpj44f8fh2/ansible-posix-1.6.0-a05q4eg_
Installing 'ansible.posix:1.6.0' to '/tmp/collections-oZe/ansible_collections/ansible/posix'
ansible.posix:1.6.0 was installed successfully
Downloading https://galaxy.ansible.com/api/v3/plugin/ansible/content/published/collections/artifacts/fedora-linux_system_roles-1.88.9.tar.gz to /root/.ansible/tmp/ansible-local-7695t4x0lor3/tmpj44f8fh2/fedora-linux_system_roles-1.88.9-wziojtcx
Installing 'fedora.linux_system_roles:1.88.9' to '/tmp/collections-oZe/ansible_collections/fedora/linux_system_roles'
fedora.linux_system_roles:1.88.9 was installed successfully
Downloading https://galaxy.ansible.com/api/v3/plugin/ansible/content/published/collections/artifacts/containers-podman-1.15.4.tar.gz to /root/.ansible/tmp/ansible-local-7695t4x0lor3/tmpj44f8fh2/containers-podman-1.15.4-bxqh_9mq
Installing 'containers.podman:1.15.4' to '/tmp/collections-oZe/ansible_collections/containers/podman'
containers.podman:1.15.4 was installed successfully
Downloading https://galaxy.ansible.com/api/v3/plugin/ansible/content/published/collections/artifacts/community-general-9.4.0.tar.gz to /root/.ansible/tmp/ansible-local-7695t4x0lor3/tmpj44f8fh2/community-general-9.4.0-ko55ql89
Installing 'community.general:9.4.0' to '/tmp/collections-oZe/ansible_collections/community/general'
community.general:9.4.0 was installed successfully
```

CS 9:
```
:: [ 14:57:40 ] :: [  BEGIN   ] :: Running 'ansible-galaxy collection install -p /tmp/collections-bTy -vv -r /tmp/postfix-lHI/meta/collection-requirements.yml'
[WARNING]: The specified collections path '/tmp/collections-bTy' is not part of
the configured Ansible collections paths
'/root/.ansible/collections:/usr/share/ansible/collections'. The installed
collection will not be picked up in an Ansible run, unless within a playbook-
adjacent collections directory.
ansible-galaxy [core 2.17.4]
  config file = None
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.12/site-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible-galaxy
  python version = 3.12.5 (main, Aug 23 2024, 00:00:00) [GCC 11.5.0 20240719 (Red Hat 11.5.0-2)] (/usr/bin/python3.12)
  jinja version = 3.1.4
  libyaml = True
No config file found; using defaults
Starting galaxy collection install process
Process install dependency map
Starting collection install process
Downloading https://galaxy.ansible.com/api/v3/plugin/ansible/content/published/collections/artifacts/ansible-posix-1.6.0.tar.gz to /root/.ansible/tmp/ansible-local-60811bhkgcg1/tmpvel04m4t/ansible-posix-1.6.0-vk20ys5k
Installing 'ansible.posix:1.6.0' to '/tmp/collections-bTy/ansible_collections/ansible/posix'
ansible.posix:1.6.0 was installed successfully
Downloading https://galaxy.ansible.com/api/v3/plugin/ansible/content/published/collections/artifacts/fedora-linux_system_roles-1.88.9.tar.gz to /root/.ansible/tmp/ansible-local-60811bhkgcg1/tmpvel04m4t/fedora-linux_system_roles-1.88.9-6lw41x5r
Installing 'fedora.linux_system_roles:1.88.9' to '/tmp/collections-bTy/ansible_collections/fedora/linux_system_roles'
fedora.linux_system_roles:1.88.9 was installed successfully
Downloading https://galaxy.ansible.com/api/v3/plugin/ansible/content/published/collections/artifacts/containers-podman-1.15.4.tar.gz to /root/.ansible/tmp/ansible-local-60811bhkgcg1/tmpvel04m4t/containers-podman-1.15.4-z4kvwlw5
Installing 'containers.podman:1.15.4' to '/tmp/collections-bTy/ansible_collections/containers/podman'
containers.podman:1.15.4 was installed successfully
Downloading https://galaxy.ansible.com/api/v3/plugin/ansible/content/published/collections/artifacts/community-general-9.4.0.tar.gz to /root/.ansible/tmp/ansible-local-60811bhkgcg1/tmpvel04m4t/community-general-9.4.0-zyaa6k6b
Installing 'community.general:9.4.0' to '/tmp/collections-bTy/ansible_collections/community/general'
community.general:9.4.0 was installed successfully
```

All installing to specified directories.

ANSIBLE_COLLECTIONS_PATH(S) is set in the func rolesGetCollectionPath

Reply:
Ok - looks like it is working here - for some reason it doesn't work with `tox-lsr` `runqemu.py`